bugbug/bug_features.py

```diff
@@ -434,63 +434,87 @@ def __init__(
     def fit(self, x, y=None):
         return self
 
+    # in case of duplicate model, bugs is a vector of tuples
```

Comment:
The changes here are too specific to the duplicate mode, we should make them more generic for any future model that might want to use a couple of bugs as a trainable item.
I'll see if I can come up with some idea tomorrow.

Reply:
One way can be to use a `flag` variable as an attribute (set to False initially) in the base Model. The models that use tuples of bugs at a time can set flag to True in their constructors. We can then pass this as an argument to BugExtractor. This way we can be sure that models with tuples always have self.flag= False and have a common set of extractors for them.

Reply:
`flag` is definitely not a great name for an option, as it could mean basically anything :P

Anyway, this doesn't change the current situation much, as we could already automatically set the flag in BugExtractor simply by checking whether the elements of bugs are dicts or tuples.

Reply:
Here's what we could do:
```
    def transform(self, elems):
        results = []

        def apply_transform(bug):
            # Do all we need to do here

            return {
                "data": data,
                "title": bug["summary"],
                "first_comment": bug["comments"][0]["text"],
                "comments": " ".join([c["text"] for c in bug["comments"]]),
            }

        for elem in elems:
            if isinstance(elem, dict):
                results.append(apply_transform(elem))
            elif isinstance(elem, tuple):
                bug1, bug2 = elem
                result1 = apply_transform(bug1)
                result2 = apply_transform(bug2)
                results.append({
                    "data1": result1["data"],
                    "data2": result2["data"],
                    "title1": result1["title"],
                    "title2": result2["title"],
                    ...
                })
            else:
                assert False, 'Unexpected element type'

        return pd.DataFrame(results)
```

Reply:
Then, in the model, you somehow merge title and first_comment of the two bugs, either with a built-in transformer if you find one that would work here, or with a custom transformer.

Reply:
> Then, in the model, you somehow merge title and first_comment of the two bugs, either with a built-in transformer if you find one that would work here, or with a custom transformer.

Or, for the time being, you can add a `merge_data` option to BugExtractor do:
```
if merge_data:
                results.append({
                    "text": f"{result1['title']} {result1['first_comment']} {result2['title']} {result2['first_comment']}",
                })
else:
                results.append({
                    "data1": result1["data"],
                    "data2": result2["data"],
                    "title1": result1["title"],
                    "title2": result2["title"],
                    ...
                })
```
instead of:
```
                results.append({
                    "data1": result1["data"],
                    "data2": result2["data"],
                    "title1": result1["title"],
                    "title2": result2["title"],
                    ...
                })
```